nrf_wifi/fw_if/umac_if/src/event.c

```diff
@@ -970,8 +978,40 @@ static enum nrf_wifi_status umac_process_sys_events(struct nrf_wifi_fmac_dev_ctx
 					def_dev_ctx->vif_ctx[mode_event->if_index]->txinjection_mode
 									= true;
 #endif /* CONFIG_NRF700X_RAW_DATA_TX */
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+					def_dev_ctx->vif_ctx[mode_event->if_index]->promisc_mode
+									= false;
+#endif /* CONFIG_NRF700X_PROMISC_DATA_RX */
 				}
 #endif /* CONFIG_NRF700X_RAW_DATA_RX */
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+				else if (mode_event->op_mode == (NRF_WIFI_STA_MODE |
+								 NRF_WIFI_PROMISCUOUS_MODE |
+								 NRF_WIFI_TX_INJECTION_MODE)) {
+					def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
+								NRF_WIFI_STA_PROMISC_TX_INJECTOR;
```

Comment:
Shouldn't this also be protected by `#ifdef CONFIG_NRF700X_RAW_DATA_TX`

Reply:
No, the interface types and modes come from UMAC header file. Setting mode to Station + Promiscuous + TX injection has a checks. This block does not need that check as it will only be hit if the mode is able to be set to this.